calicoctl/commands/node/status.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/docker/engine-api/client"
+	"github.com/docker/engine-api/types"
+	"github.com/docopt/docopt-go"
+	"golang.org/x/net/context"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+var (
+	ipv4 = true
+
+	// Check for Word_<IP> where every octate is seperated by "_", regardless of IP protocols
+	// Example match: "Mesh_192_168_56_101"
+	bgpPeerRegex, _ = regexp.Compile(`\w+_\d+_\d+_\d+_\d+`)
+)
+
+// Status prings status of the node and returns error (if any)
+func Status(args []string) error {
+	doc := `Usage:
+calicoctl node status
+
+Description:
+  Display the status of the Calico node`
+
+	_, _ = docopt.Parse(doc, args, true, "calicoctl", false, false)
+
+	ctx := context.Background()
+
+	defaultHeaders := map[string]string{"User-Agent": "engine-api-cli-1.0"}
+	cli, err := client.NewClient("unix:///var/run/docker.sock", "v1.22", nil, defaultHeaders)
+	if err != nil {
+		return err
+	}
+
+	options := types.ContainerListOptions{All: true}
+	containers, err := cli.ContainerList(ctx, options)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, c := range containers {
+		if strings.Contains(c.Names[0], "calico-node") && c.State == "running" {
+
+			fmt.Printf("calico-node container is %s. Status: %s\n", c.State, c.Status)
```

Comment:
Hmm.. yeah I guess this was a bit silly (since c.State will _always_ be running).

I had originally though we could show non-running state as well, but there might be a bunch of exited calico-node containers that we don't care about (so long as one is running), so the way you had it is probably best... sorry!